Doctrine/Orm/Filter/OrderFilter.php

```diff
@@ -68,17 +70,35 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder)
         }
 
         $properties = $this->extractProperties($request);
-        $fieldNames = array_flip($this->getClassMetadata($resource)->getFieldNames());
 
         foreach ($properties as $property => $order) {
-            if (!$this->isPropertyEnabled($property) || !isset($fieldNames[$property])) {
+            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resource)) {
                 continue;
-            } elseif ('' === $order && isset($this->properties[$property])) {
+            }
+
+            if ('' === $order && isset($this->properties[$property])) {
                 $order = $this->properties[$property];
             }
 
             $order = strtoupper($order);
-            if ('ASC' === $order || 'DESC' === $order) {
+            if (!in_array($order, ['ASC', 'DESC'])) {
+                continue;
+            }
+
+            if ($this->isPropertyNested($property)) {
+                $propertyParts = $this->splitPropertyParts($property);
+
+                $alias = null;
+                $parentAlias = 'o';
+
+                foreach ($propertyParts['associations'] as $association) {
+                    $alias = sprintf('%s_%s', self::DQL_NAMESPACE, $association);
+                    $queryBuilder->join(sprintf('%s.%s', $parentAlias, $association), $alias);
```

Comment:
So if you got several properties with the same parent you will end up doing the join several times right?